backend/api/serializers/computetask.py

```diff
@@ -53,15 +55,45 @@ class Meta:
 
 
 class ComputeTaskOutputSerializer(serializers.ModelSerializer, SafeSerializerMixin):
+    ASSET_SERIALIZERS = {
+        ComputeTaskOutputAsset.Kind.ASSET_MODEL: 0,
+        ComputeTaskOutputAsset.Kind.ASSET_PERFORMANCE: 0,
+    }
+
     class Meta:
         model = ComputeTaskOutput
         fields = [
             "identifier",
             "permissions",
             "transient",
+            "value",
         ]
 
     permissions = make_download_process_permission_serializer()(source="*")
+    value = serializers.SerializerMethodField(source="*", read_only=True)
+
+    def get_value(self, task_output):
+        if task_output.task.status != ComputeTask.Status.STATUS_DONE:
+            return None
+
+        data = []
+        for output_asset in task_output.assets.all():
+            if output_asset.asset_kind == ComputeTaskOutputAsset.Kind.ASSET_MODEL:
+                model = Model.objects.get(key=output_asset.asset_key)
+                data.append(ModelSerializer(instance=model).data)
+            elif output_asset.asset_kind == ComputeTaskOutputAsset.Kind.ASSET_PERFORMANCE:
+                task_key, metric_key = output_asset.asset_key.split("|")
+                perf = Performance.objects.get(compute_task__key=task_key, metric__key=metric_key)
+                data.append(perf.value)
+
+        # FIXME: we should better always return a list,
+        # but it may requires some adapations on the frontend side
+        if len(data) == 0:
+            return None
+        elif len(data) == 1:
+            return data[0]
+        else:
+            return data
```

Comment:
Do you think we can always return a list instead?

Reply:
This would break the front, so we'll need frontend PR to be ready before merging this one. But I guess that should be feasible. @HamdyD could you check with this PR to see if we can handle output value always as list ?

Reply:
@alexisdeh I believe there were some discussions on this subject, do you remember why we chose to return None or single value when relevant instead of always a list that could be empty or of size of 1 ?

Reply:
It was how the api was designed:
1/ If the task is not done there is no value at all for the output, so `None` is set. This is different than an empty list that could be a proper output value. The choice was to use 'None' only to explicit there is no value.
2/ The type of the value field is polymorphic anyway (it could be for example a model, or a performance, depending on the `asset_kind` property of the output in the algo). So this polymorphic approach has been embraced, and the type also depends on the `multiple` property of the output: Only in case it is multiple then we return a list.
This is totally debatable, and can be changed.